src/filemap.rs

```diff
@@ -36,16 +36,95 @@ pub fn write_all_files(file_map: &FileMap,
                        config: &Config)
                        -> Result<(HashMap<String, String>), io::Error> {
     let mut result = HashMap::new();
+    output_heading(mode).ok();
+
     for filename in file_map.keys() {
         let one_result = try!(write_file(&file_map[filename], filename, mode, config));
         if let Some(r) = one_result {
             result.insert(filename.clone(), r);
         }
     }
+    output_trailing(mode).ok();
 
     Ok(result)
 }
 
+pub fn output_heading(mode: WriteMode) -> Result<(), io::Error> {
```

Comment:
Put output_heading and output_trailing inside write_file, we're already matching write_mode there, no need to take in a mode parameter and match again. Also, use try! to propagate the error instead of silently ignore it.